### ISSUE TITLE
JDK 1.8 Update 20

### ISSUE THREAD

**Issue by pejax** [salat/salat#133]
I updated to JDK 1.8.0_20 from JDK 1.8.0_11 on OS-X 10.9.4 today and my Scala Play 2.2.3 project crashes at startup now. I reverted to 1.8.0_11. Could you take a look  please?

scala.tools.scalap.scalax.rules.ScalaSigParserError: Unexpected error: Not a valid class file
     scala.tools.scalap.scalax.rules.Rules$$anonfun$expect$1.apply(Rules.scala:76)
     scala.tools.scalap.scalax.rules.scalasig.ClassFileParser$.parse(ClassFileParser.scala:97)
     com.novus.salat.util.ScalaSigUtil$.parseClassFileFromByteCode(ScalaSigUtil.scala:36)
     com.novus.salat.util.ScalaSigUtil$.parseScalaSig0(ScalaSigUtil.scala:79)
     com.novus.salat.ClassAnalyzer$.findSym(ClassAnalyzer.scala:48)
     com.novus.salat.ClassAnalyzer.<init>(ClassAnalyzer.scala:136)
     com.novus.salat.ConcreteGrater.ca$lzycompute(Grater.scala:118)
     com.novus.salat.ConcreteGrater.ca(Grater.scala:118)
     com.novus.salat.ConcreteGrater.asObject(Grater.scala:294)
     com.novus.salat.dao.SalatDAO$$anonfun$findOne$1.apply(SalatDAO.scala:349)
     com.novus.salat.dao.SalatDAO$$anonfun$findOne$1.apply(SalatDAO.scala:349)
     scala.Option.map(Option.scala:145)
     com.novus.salat.dao.SalatDAO.findOne(SalatDAO.scala:349)
     com.novus.salat.dao.ModelCompanion$class.findOne(ModelCompanion.scala:176)
     models.Admin$.findOne(Admin.scala:15)
     com.novus.salat.dao.BaseDAOMethods$class.findOne(DAO.scala:116)
     models.Admin$.findOne(Admin.scala:15)
     models.Admin$.findByEmail(Admin.scala:23)
     Global$$anonfun$onStart$1$$anonfun$apply$1.apply(Global.scala:20)
     Global$$anonfun$onStart$1$$anonfun$apply$1.apply(Global.scala:18)
     scala.Option.map(Option.scala:145)
     Global$$anonfun$onStart$1.apply(Global.scala:18)
     Global$$anonfun$onStart$1.apply(Global.scala:17)
     scala.collection.immutable.List.flatMap(List.scala:327)
     Global$.onStart(Global.scala:17)
     play.api.GlobalPlugin.onStart(GlobalSettings.scala:220)
     play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:91)
     play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:91)
     scala.collection.immutable.List.foreach(List.scala:381)
     play.api.Play$$anonfun$start$1.apply$mcV$sp(Play.scala:91)
     play.api.Play$$anonfun$start$1.apply(Play.scala:91)
     play.api.Play$$anonfun$start$1.apply(Play.scala:91)
     play.utils.Threads$.withContextClassLoader(Threads.scala:21)
     play.api.Play$.start(Play.scala:90)
     play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:157)
     play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:130)
     scala.Option.map(Option.scala:145)
     play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:130)
     play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:128)
     scala.util.Success.flatMap(Try.scala:230)
     play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:128)
     play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:120)
     scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
     scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
     scala.concurrent.forkjoin.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1361)
     scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
     scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
     scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
     scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

**Comment by anfuerer** [comment]
We're also seeing this issue in the following setup:
* Scala 2.11.5
* JRE build 1.8.0_25-b17
* HotSpot build 25.25-b02 (mixed mode)

However, interesting is that everything is running with Scala 2.10.4 and the same 1.8 JVM.

**Comment by rktoomey** [comment]
Is 2.11.4 working in this setup?

**Comment by anfuerer** [comment]
Unfortunately, can't test. Compilation of our project with 2.11.4 doesn't terminate due to https://issues.scala-lang.org/browse/SI-9032

> Is 2.11.4 working in this setup?

**Comment by anfuerer** [comment]
I just had the issue with scala 2.10.4 and the 1.8 JVM as well. So, please ignore my statement in the comment above.

**Comment by gabeos** [comment]
Not sure if this'll be useful, but I had the same bug under different conditions. I'm running under Intellij IDEA 13, JDK 1.8.0_25, with Scala 2.11.2.

Everything works just fine when I run the code, but when I try to debug the code (with IDEA), the exact same code fails consistently when trying to deserialize a case object.
The case object is extremely simple, just something used as a placeholder in the object when some data isn't know, and contains only a string, although it is pretty deeply nested in a larger object.

I have the bytecode file of the case object in question, as well as the error output here: 
(through the intellij "View Bytecode" plugin.): https://gist.github.com/gabeos/1e90dc497a49e5eea389

Let me know if there's any other info I can provide that might help you track this one down.

**Comment by sief** [comment]
I just noticed that I can reproduce this only when play is run in dev mode (activator run), so far no problems in prod mode (activator start). Tests are also ok. Maybe this is related to play's hot code reloading.

Environment:
OSX 10.10.1 
java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
Scala 2.11.5
sbt 0.13.7
play 2.3.7

**Comment by pejaham** [comment]
Tried again with JDK1.8.0_40 - seems to be back working.

**Comment by wajda** [comment]
Yes, it works fine now. So it was a Java bug and it's fixed in the recent updates.
